From 2f400027866fb1d30b066c6acf6930b2dd50f266 Mon Sep 17 00:00:00 2001
From: Bo-Cun Chen <bc-bocun.chen@mediatek.com>
Date: Mon, 22 Sep 2025 11:12:14 +0800
Subject: [PATCH] net: ethernet: mtk_eth_soc: record both the GMAC and XGMAC
 speeds.

Without this patch, the ETH driver is unable to record the XGMAC speed
in the mtk_mac structure.

Signed-off-by: Bo-Cun Chen <bc-bocun.chen@mediatek.com>
---
 drivers/net/ethernet/mediatek/mtk_eth_soc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/mediatek/mtk_eth_soc.c b/drivers/net/ethernet/mediatek/mtk_eth_soc.c
index d5e0d9e..3df84e2 100644
--- a/drivers/net/ethernet/mediatek/mtk_eth_soc.c
+++ b/drivers/net/ethernet/mediatek/mtk_eth_soc.c
@@ -1005,7 +1005,6 @@ static void mtk_gdm_mac_link_up(struct mtk_mac *mac,
 		 MAC_MCR_FORCE_RX_FC);
 
 	/* Configure speed */
-	mac->speed = speed;
 	switch (speed) {
 	case SPEED_2500:
 	case SPEED_1000:
@@ -1099,6 +1098,7 @@ static void mtk_mac_link_up(struct phylink_config *config,
 	struct mtk_mac *mac = container_of(config, struct mtk_mac,
 					   phylink_config);
 
+	mac->speed = speed;
 	if (mtk_is_netsys_v3_or_greater(mac->hw) && mtk_interface_mode_is_xgmii(interface))
 		mtk_xgdm_mac_link_up(mac, phy, mode, interface, speed, duplex,
 				     tx_pause, rx_pause);
-- 
2.45.2

